doc_analyzer.py

```diff
@@ -145,25 +144,47 @@ def get_macros_infos(self):
 
         macros_infos = [
             {'number': vbaparser.nb_autoexec,
-                'description': 'Ключевые слова автоматического вызова', 'danger': False},
+                'description': 'Ключевые слова автоматического вызова', 'danger': False, 'function': olevba.detect_autoexec},
             {'number': autoexec_and_HEX,
-                'description': 'Автоматический вызов шеснадцатиричных обфусцированных строк', 'danger': False},
+                'description': 'Автоматический вызов шеснадцатиричных обфусцированных строк', 'danger': False, 'function': olevba.detect_hex_strings},
             {'number': vbaparser.nb_vbastrings,
-                'description': 'VBA обфусцированные строки', 'danger': False},
+                'description': 'VBA обфусцированные строки', 'danger': False, 'function': olevba.detect_vba_strings},
             {'number': vbaparser.nb_suspicious,
-                'description': 'Подозрительные ключевые слова', 'danger': False},
+                'description': 'Подозрительные ключевые слова', 'danger': False, 'function': olevba.detect_suspicious},
             {'number': autoexec_and_vba,
-                'description': 'Автоматический вызов обфусцированного кода', 'danger': True},
+                'description': 'Автоматический вызов обфусцированного кода', 'danger': True, 'function': olevba.detect_vba_strings},
             {'number': vbaparser.nb_dridexstrings,
-                'description': 'Dridex обфусцированные строки', 'danger': True},
+                'description': 'Dridex обфусцированные строки', 'danger': True, 'function': olevba.detect_dridex_strings},
             {'number': autoexec_and_base64,
-                'description': 'Автоматический вызов Base64 обфусцированных строк', 'danger': True},
+                'description': 'Автоматический вызов Base64 обфусцированных строк', 'danger': True, 'function': olevba.detect_base64_strings},
         ]
 
         vbaparser.close()
         return macros_infos
 
 
+    def get_vba_code(self):
+        vbaparser = VBA_Parser(self.file_path)
+        vbaparser.detect_vba_macros()
+        code_list = list()
+
+        for (filename, stream_path, vba_filename, vba_code) in vbaparser.extract_macros():
+            code_list.append(vba_code)
+            code = vba_code
+
+        report_vba = code_list[0]
```

Comment:
No null check first? Basically it fails if file has no macros.
Also some message, i.e. no macros found?

Reply:
ok, I agree